Drop stale Python versions, test only on new ones

Toshakins opened the pull request:
Hey, it's time to upgrade! Very minor code changes, just want to be sure everything runs on Py3.6+. Please let me know if any further work needed, I'm willing to commit my efforts. Thanks!

Toshakins committed:
- Drop stale Python versions, test only on new ones

Toshakins committed:
maybe do not bump if no code changes? also, trigger trvis build

Toshakins committed:
remove code related to python 2.6

thomasw commented:
@Toshakins, thanks for the contribution. Can you make a pass at updating all of the dev dependencies to the latest versions? You'll need to power through that to get the test suite to run and then we can go from there.

thomasw committed:
Update development dependencies to latest versions

thomasw committed:
Remove unused file

thomasw committed:
setup.py: Simplify test requirements

This removes selective requirement installation for python <3 as it is no longer needed.

thomasw commented:
I spun up a branch for you that has these changes: https://github.com/thomasw/querylist/compare/update-dev-dependencies?expand=1

thomasw committed:
setup.py: update trove classifiers

thomasw committed:
setup.py: add a long description

Set the package's long description to the contents of the readme.

thomasw committed:
setup.py: set license to MIT

Toshakins committed:
Merge remote-tracking branch 'thomasw/update-dev-dependencies'

Toshakins commented:
Sorry for those broken tests. Travis was queuing the build for a while, and I've lost patience. Merged the branch you mentioned, tests pass on py3.6, waiting for Travis.

Toshakins committed:
cheating to pass the tests

Toshakins commented:
Tests fail for pypy interpreter. Would it be cheating to make this library available for pypy3 only?

thomasw commented:
> Tests fail for pypy interpreter. Would it be cheating to make this library available for pypy3 only?

Not cheating at all and totally inline with the other changes. Thanks.

thomasw commented on tests/base.py:
```diff
@@ -1,4 +1 @@
-try:
```
We can eliminate this file now and directly import TestCase in the test files.

Toshakins replied:
As a consequence, I re-ordered imports in `tests` folder. Not sure if you have a strong opinion on imports, but anyway I'm okay to change them if needed.

thomasw commented on setup.py:
```diff
@@ -29,12 +18,22 @@
+        'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8',
+        'Programming Language :: Python :: 3.9',
+        'Programming Language :: Python :: Implementation :: PyPy'
```
Ugh, let's also add

Programming Language :: Python :: Implementation :: CPython

thomasw commented on README.md:
```diff
@@ -26,7 +26,7 @@ Querylist can be installed like any other python package:
     > pip install querylist
 
-Querylist is tested against Python 2.6, 2.7, 3.3, 3.4, and pypy.
+Querylist is tested against Python 3.6+ and pypy.
```
There might be duplicate content in docs/ that needs updating.

Toshakins replied:
Thanks, updated docs too. Also, I allowed myself to update CHANGELOG.

thomasw commented:
@Toshakins I'm assuming you're making updates here to ship a python3 upgrade for something Yola related?

What's your internal timeline looking like? Once we get this merged, I may try to get a little more cleanup in before shipping a new release.

I encourage you to rigorously verify that whatever depends on this in a python3 environment continues to work as expected. I haven't used this library myself in a while and I'm kinda just moving forward here assuming that the test suite I remember nothing about is sufficient to verify that things work as expected.

Also, please tell V that I said hello!

Toshakins commented:
> I'm assuming you're making updates here to ship a python3 upgrade for something Yola related?

Yes! You're right. We are migrating to Python 3.6, actually.

> What's your internal timeline looking like?

We tested the latest available version, and it worked, so we decided to ship with 0.4.0. But next generations of Yola people might be extremely confused by that decision. The project to upgrade Querylist does not have any schedule, really. It is my initiative, and I was hoping the whole upgrade won't take much effort. On the other hand, you didn't respond to the pull request we would consider forking and upgrading :) Feel free to do any additional cleanup, there are no time frames for that.

> Also, please tell V that I said hello!

Nice! He hellos(is it a word?) you back!

Toshakins committed:
Import unittest directly and other small fixes

Toshakins committed:
update changelog.rst

Toshakins commented:
I've applied fixes for the comments above.

thomasw commented:
I'm hoping to get to this over the weekend. I have a branch where I've rewritten some of your history and bundled some related cleanup. Please @mention me if you make additional changes here.

Toshakins commented:
Hello again! Caring about repositories could be hard. Huge thanks that you were able to take a look into my PR. Is it possible to publish a new version? It's ok if you have little time for that. I wanted only to follow up and do not want to push the release.
